### PR TITLE
Making tests not depend on execution order

### PR DESCRIPTION
**Pull request by fleupold**
We were using the `before` hook to set up our state. Since we never reset the state, this means that our tests are interdependent on each other, meaning that reordering them or taking out a test could change the entire behaviour.

This is very brittle and not good practice. State should be cleaned between each test and not be dependent on each other.

This change achieves this by using mock instead of actually deployed contracts and most importantly creating the contract under test anew in each test.

### CODE REVIEW

**Review by josojo — APPROVED** [review]
At some point, we might wanna clean up the tests a little and collect all deposit  / exit-deposits in one separate js file.
Like the mock-up testing method.

File: test/plasma.js [hunk 1]
```diff
@@ -64,8 +56,7 @@ contract('Plasma', (accounts) => {
     })
 
     it('Rejected: Wrong amount (root != depositHash)', async () => {
-      // WEIRD PROBLEM - TODO: This test doesn't pass with with oneETH + 1!
```
Comment (bh2smith) [inline test/plasma.js:67]: I don't think this should be removed unless you have solved the problem stated here.

**Review by bh2smith — CHANGES_REQUESTED** [review]
Everything appears to be great. I am also very glad to see the use of MockContracts! I really don't think you should remove that TODO, unless you have TODONE it.

File: test/plasma.js (hunk 1 above)
Comment (fleupold) [reply test/plasma.js:67]: The reason is that in js `(10**18) + 1 === (10**18)` evaluates to true. So oneETH + 1 is the same as oneETH. I think using a significantly different amount is testing what we are trying to test.

Comment (bh2smith) [reply test/plasma.js:67]: That is a very strange property of integers that JS has. Anyway, then as long at its a precision problem with JS, then I suppose its fine to merge.

**Review by bh2smith — APPROVED** [review]

**Pull request merged**